src/test/java/spoon/test/refactoring/MethodsRefactoringTest.java

```diff
@@ -73,4 +89,117 @@ private void checkContainsOnly(List<String> foundNames, String... expectedNames)
 		}
 		assertTrue("Unexpected names found: "+foundNames, foundNames.isEmpty());
 	}
-}
+
+	@Test
+	public void testAllMethodsSameSignatureFunction() {
+		final Launcher launcher = new Launcher();
+		launcher.setArgs(new String[] {"--output-type", "nooutput","--level","info" });
+		launcher.getEnvironment().setCommentEnabled(true);
+		launcher.addInputResource("./src/test/java/spoon/test/refactoring/parameter/testclasses");
+		launcher.run();
+		Factory factory = launcher.getFactory();
+		
+		//contract: check that found methods does not depend on the starting point
+		checkMethodHierarchy("A_method1", factory.Class().get(TypeA.class).getMethodsByName("method1").get(0));
+		checkMethodHierarchy("A_method1", factory.Class().get(TypeC.class).getMethodsByName("method1").get(0));
+		checkMethodHierarchy("A_method1", factory.Interface().get(IFaceB.class).getMethodsByName("method1").get(0));
+		checkMethodHierarchy("A_method1", factory.Class().get(TypeL.class).getMethodsByName("method1").get(0));
+		checkMethodHierarchy("A_method1", ((CtClass<?>)factory.Class().get(TypeB.class).filterChildren(new NameFilter<>("1Local")).first()).getMethodsByName("method1").get(0));
+		checkMethodHierarchy("R_method1", factory.Class().get(TypeR.class).getMethodsByName("method1").get(0));
+		checkMethodHierarchy("R_method1", factory.Class().get(TypeS.class).getMethodsByName("method1").get(0));
+	}
+	
+	private void checkMethodHierarchy(String hierarchyName, CtMethod startMethod) {

```

Comment:
as far as I understand, the first parameter is a kind of expected value.

what about 
`checkMethodHierarchy(String hierarchyName, CtMethod startMethod)` -> `assertAllMethodsSameSignatureFunction(Collection<Methoc> expected,  CtMethod startMethod)`

called this way `assertAllMethodsSameSignatureFunction(collectMethodsInHierarchy(AMethod1TypeHiearachy.class),  startMethod)`